### PR TITLE
wherein I try to make shitcode not shit but unfortunately it quickly proves to only bring me wracking grief and impotent frustration at multiple things at once so I end up not doing that

### LINKED ISSUE #7511: Botany3: Nettles, deathnettles, novaflowers can't be grinded
Putting them into grinder and grinding them results in 0 chems.
Novaflowers have to be put in using a plantbag.

### LINKED ISSUE #7687: Washing machine
(WEB REPORT BY: rain7x REMOTE: 158.69.55.2:7777)

> Revision (Should be above if you're viewing this from ingame!)
> e69e9243ef3ee2894ef0a867d224ecd06fd85d17 
> 
> General description of the issue
> The washing machine when clicked on while it is washing clothing says "The the washing machine is busy."
> 
> What you expected to happen
> It to say "The washing machine is busy."
> 
> What actually happened
> It says "The the washing machine is busy."
> 
> Steps to reproduce if possible
> Put clothes in washing machine, left click while it is still washing

### PR DESCRIPTION
**Pull request by 9600bauds**
adds "empty soda can" verb
does some mortar bullshit i honestly forgot what
fixes #7511 (would have been useful if the report mentioned this bug was only for mortars in the first place)
fixes #7687 because I am a huge fucking autist apparently

i struggled with this horrible, horrible whitespace bullshit where all of git was collectively throwing errors at me and I lost some changes at which point I couldn't even roll back the whole thing because holy jesus FUCK the only thing worse than esoteric git awfulness that has no reason for existing is esoteric git awfulness that has no reason for existing and entirely orbits around changes that are literally not there and do not matter and all of which is happening while you're kind of drunk so fuck this whole thing I am not unshittening grinders, they remain the shittiest code I have seen, I tried to fix it and I gave up and so will you

### CODE REVIEW

**Comment by Exxion** [comment]
See now I want to look at grinder code but I also know I will want to die if I do

**Comment by Kurfursten** [comment]
Grinder code is some of the worst. Mortar code isn't great but it isn't as bad (which is not a feat, that just means it's not unreadably bad).

**Comment by Shadowmech88** [comment]
How could grinder code be so bad? The concept seems fairly simple: delete the items inside and deposit the relevant reagents in the beaker.
It seems like such a simple thing to code. Or are the workings of the grinder more complex than I think they are?

**Comment by Kurfursten** [comment]
@Shadowmech88 Some things get snowflake behaviors, for example 

> In most cases, put the reagent contained within into the beaker
> If it's on a special list of items, disregard what reagents it contains and instead give us a different reagent (wheat -> flour, tomato -> ketchup, seeds -> blackpepper, metal sheets -> iron, etc.)
> If it's on a special list of foods AND juice was selected, disregard reagents and instead give us juice (melon slices, apples, lemons, etc.)
> If it's a xenoarch rock you have to also transfer data to the new crushed reagent

But wait, there's more. Some of the snowflake give set amounts of their reagents (example: wheat always gives 5 flour) whereas some snowflake stuff gives an amount based on how much nutriment is inside (example: you get more corn oil by growing corn with more nutriment (higher potency) or ... injecting the food with nutriment before you grind it)

And on top of everything else weird about this, grinder is written in such a way as to be nearly completely obfuscated about what it's doing with uncommented code, unidentified numbers, and unlabelled proc calls. The mortar keeps the snowflakey behaviors of the grinder, but presents it in a more readable way.

**Comment by Exxion** [comment]
You'd be amazed how bad oldcode can be
BYOND allows new and innovative ways to write horrifyingly bad code

**Comment by 9600bauds** [comment]
I am sorry kurfursten, I really did not want to say it but the mortar was pretty bad as well. It still uses hard coded amounts of chemicals per grown item. Whereas you can inject nutriment into corn to blend it for more oil, you can also just mortar it to get 3 times as more. It also constantly tried to add a reagent that didn't exist.

**Comment by Kurfursten** [comment]
The mortar is like that so that its outputs will be identical to the grinder - syntax can be improved but I had to keep the snowflake behaviors. The choice was to either have the grinder and mortar produce different amounts or to rewrite the grinder completely which was out of scope of the PR and would've affected hydroponics/kitchen balance.

(You're committing the same sin of sloth here if you don't re-write both of them entirely now that you're touching the file, though)

**Comment by 9600bauds** [comment]
I DID.
IT WAS ABSOLUTE ASS AND I REGRET EVERY MOMENT OF IT AND I LOST ALL THE CHANGES I MADE TO THE ESOTERIC GIT BULLSHIT THAT HAS BEEN PLAGUING MY EXISTENCE FOR THE PAST FIVE DAYS AND I WILL NOT DO IT AGAIN.

...Also the grinder and the mortar do produce different amounts, precisely because it doesn't keep the snowflake behaviors but also because it flat-out has different amounts hardcoded in

**Comment by Kurfursten** [comment]
Weird. It's not supposed to do that. Nobody ever reported it, although I know for sure it gets the right amount for metal sheets.

You can't restore?

**Comment by 9600bauds** [comment]
patch does not apply
patch does not apply
.git/rebase-apply/patch:257: trailing whitespace.
changes:
warning: seven hundred thousand lines add whitespace errors.
Falling back to patching base and 3-way merge...
error: Your local changes to the following files would be overwritten by merge:
All of your files changed plus forty more you didn't change

**Comment by Kurfursten** [comment]
Okay just ditch your changes to chemistry-machinery or close this PR and I'll re-write grinder and mortar today.

Besides, I'd rather empty soda cans by clicking the floor to splash than use a verb.

**Comment by 9600bauds** [comment]
Can't empty soda cans by clicking on the floor without heinous copypaste because for some awful reason they're children of food, not reagentcontainers or reagentcontainers/glass. Also I think I added something to mortars so that you can transfer reagents properly into them with syringes and droppers and stuff.

**Comment by Kurfursten** [comment]
Move splash code up to reagent_containers and have it check for OPENCONTAINER

**Comment by Shadowmech88** [comment]
> children of food, not reagentcontainers

Isn't all food a child of reagent_containers?

**Comment by Kurfursten** [comment]
@Shadowmech88 It is, he meant glass which is where splashing is handled (line 85), it's easy to move up another level though

**Comment by 9600bauds** [comment]
I meant that all drinks are a child of /obj/item/weapon/reagent_containers/food/ for some unknowable reason, and soda cans are a child of /obj/item/weapon/reagent_containers/food/drinks/.

**Comment by Kurfursten** [comment]
@9600bauds So that they inherit the behavior to transfer reagents into your body when used in hand

File: code/modules/reagents/reagent_containers/food/drinks.dm [hunk 1]
```diff
@@ -614,6 +614,16 @@
 /obj/item/weapon/reagent_containers/food/drinks/soda_cans
 	vending_cat = "carbonated drinks"
 
+/obj/item/weapon/reagent_containers/food/drinks/soda_cans/verb/empty()
```
Comment (PJB3005) [inline code/modules/reagents/reagent_containers/food/drinks.dm:617]: This is gonna need some sanity related to the `usr`.

**Comment by 9600bauds** [comment]
vgstation13_-_ dsmartgit_reposvgstation13 _-_sm_2016-01-17_17-52-35
Just fuck this whole thing desu
I am so mad at this whitespace obscenity that my grandchildren will still be mad at it and won't know why

**Comment by Shadowmech88** [comment]
Yell at Clusterfack about it.

Just commit the "changes" and make them into a PR, that's the only way I was able to fix it.

**Comment by gbasood** [comment]
Just commit all the whitespace changes anyway. It will fuck up nothing. Everyone's going to have to deal with whitespace changes on the PRs they were working on for awhile.

**Comment by Exxion** [comment]
Just update from Bleeding-Edge
Fixed it for me with no trouble

**Comment by 9600bauds** [comment]
I literally can't commit the whitespace changes.
I can't fix the conflicts BECAUSE of the whitespace changes that I CAN'T SEE.
I already nuked the whole fucking thing, reset to bleeding-edge, copypasted the changes from the diff, and IT'S CONFLICTING FOR EVERY SINGLE PR EVEN IF IT DOESN'T TOUCH THE SAME FILES.

**Comment by gbasood** [comment]
```
git add .
git commit -m "honk"
```

**Comment by Shadowmech88** [comment]
> I literally can't commit the whitespace changes

What happens when you try?

**Comment by 9600bauds** [comment]
There's conflicts, I can't fix those conflicts and committing the whitespace changes doesn't help with that end.
This PR will soon conflict with kurfurt's grinder refactor and I honestly don't care to have this merged. I just want to forget about this whole whitespace disaster.

**Comment by Kurfursten** [comment]
Just do as I recommended before and open a new PR that moves splash code up a level and checks for OPENCONTAINER

**Comment by clusterfack** [comment]
When bauds, when

**Pull request closed without merging**